src/main/java/mekanism/common/content/sps/SPSMultiblockData.java

```diff
@@ -143,6 +167,22 @@ public boolean tick(Level world) {
         return needsPacket;
     }
 
+    private void updateCachedRecipe() {
+        ChemicalStack current = inputHandler.getInput();
+        if (current.isEmpty()) {
+            cachedRecipe = null;
+            lastCheckedInput = ChemicalStack.EMPTY;
+        } else if (cachedRecipe == null || !ChemicalStack.isSameChemical(current, lastCheckedInput)) {
+            cachedRecipe = findFirstRecipe(inputHandler);
+            if (cachedRecipe != null) {
+                lastCheckedInput = current.copy();
+                if (cachedRecipe.getOutputDefinition().getFirst().isEmpty()) {
```

Comment:
what is the point of this check? The output not being set is supposed to be handled by the `isIncomplete()` method on the recipe